src/main/java/bowling/domain/BowlingGame.java

```diff
@@ -1,43 +1,76 @@
 package bowling.domain;
 
-import bowling.domain.dto.StateDtos;
-import bowling.domain.frame.Frames;
-import bowling.domain.pin.PinCount;
 import bowling.domain.player.Player;
+import bowling.exception.message.ErrorMessage;
 
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 public class BowlingGame {
 
-    private final Player player;
-    private final Frames frames;
+    private static final int FIRST_GAME_SET_NUMBER = 0;
 
-    private BowlingGame(final Player player) {
-        this.player = player;
-        frames = Frames.newInstance();
+    private final List<GameSet> gameSets;
+    private int currentGameSetNumber;
+
+    private BowlingGame(final List<String> names) {
+        validateNames(names);
+
+        this.gameSets = names.stream()
+                .map(String::trim)
+                .map(Player::of)
+                .map(GameSet::of)
+                .collect(Collectors.toList());
+        this.currentGameSetNumber = 0;
     }
 
-    public static BowlingGame of(final Player player) {
-        return new BowlingGame(player);
+    public static BowlingGame of(final List<String> names) {
+        return new BowlingGame(names);
     }
 
-    public void play(final int hitCount) {
-        frames.bowl(PinCount.of(hitCount));
+    private void validateNames(final List<String> names) {
+        if (Objects.isNull(names)|| names.isEmpty()) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_VALUE);
+        }
     }
 
     public boolean isGameOver() {
-        return this.frames.isGameOver();
+        return gameSets.stream()
+                .allMatch(GameSet::isGameOver);
     }
 
     public String getPlayerName() {
-        return this.player.getName();
+        return this.currentGameSet().getPlayerName();
+    }
+
+    public void playOfCurrentGameSet(final int hitCount) {
+        this.currentGameSet().play(hitCount);
+    }
+
+    private GameSet currentGameSet() {
+        int index = currentGameSetNumber;
+
+        return gameSets.get(index);
+    }
+
+    public void turnOverGameSet() {
+        if (!isTurnOver()) {
+            return;
+        }
+
+        this.currentGameSetNumber++;
+
+        if (this.currentGameSetNumber == gameSets.size()) {
+            this.currentGameSetNumber = FIRST_GAME_SET_NUMBER;
+        }
```

Comment:
```suggestion
        this.currentGameSetNumber %= gameSets.size();
```
이건 그냥 참고해보세요 요런식도 가능해요 :)

Reply:
오오ㅎㅎㅎ 그렇네여!! 알고리즘할 때 저렇게 썼던 기억이 나네옄ㅋㅋㅋㅋㅋㅋ 👍